### ISSUE TITLE
Training loop problems

### ISSUE THREAD

**Issue by lsrosa** [SSIGPRO/peepholelib#166]
The default training loop is running over the complete dataset and making a `backprop` at every batch.

The default behaviour should be to run `n_iter` batches and accumulate the loss, then make one `backprop` for each epoch.
